### ISSUE TITLE
DMA crashes when a field exceeds a certain size

### ISSUE THREAD

**Issue by magnatelee** [StanfordLegion/legion#337]
The test case in commit 5d51aae kills DMA on two nodes. Here is a command to reproduce the issue on sapling:

`LAUNCHER='mpirun -np 2 -H n0000,n0001 --bind-to none -x TERRA_PATH -x LD_LIBRARY_PATH -x INCLUDE_PATH' regent.py tests/runtime/bugs/dma.rg`

According to the stacktrace, `dst_info.bytes_per_chunk` somehow got 0 and raised a divide-by-zero error. 

```
(gdb) bt
...
#4  0x00007feac210a59c in Realm::XferDes::default_get_requests (this=0x7feaa8005320, reqs=0x681ead0, nr=46, flags=2) at /home/wclee/legion/runtime/realm/transfer/channel.cc:677
#5  0x00007feac210c5b9 in Realm::RemoteWriteXferDes::get_requests (this=0x7feaa8005320, requests=0x681ead0, nr=46) at /home/wclee/legion/runtime/realm/transfer/channel.cc:1408
...
(gdb) f 4
#4  0x00007feac210a59c in Realm::XferDes::default_get_requests (this=0x7feaa8005320, reqs=0x681ead0, nr=46, flags=2) at /home/wclee/legion/runtime/realm/transfer/channel.cc:677
677                     size_t ratio = src_info.bytes_per_chunk / dst_info.bytes_per_chunk;
(gdb) p dst_info.bytes_per_chunk
$1 = 0
```

I suspect there is some hard coded threshold that this test case exceeds, because this bug disappears when I decrease the size of field or the number of elements in the region.

**Comment by lightsighter** [comment]
Can we disable this test in the test suite until we think this issue is fixed? 

https://gitlab.com/StanfordLegion/legion/-/jobs/44318147

As long as there is an open issue I think it is safe to disable the test in the suite since we know that there is someone responsible for fixing it.

**Comment by jiazhihao** [comment]
That part of code was added by Sean during his DMA merge. I will take a look at the issue tomorrow and see if I can generate an easy fix.

**Comment by magnatelee** [comment]
I just disabled the test, so CI should be passing now.

**Comment by magnatelee** [comment]
After having a discussion with Zhihao, we decided to wait for Sean to fix this. He will shortly write about what he has found.

**Comment by streichler** [comment]
Ok, I see the issue - now I just have to figure out how to fix it.  The problem has to do with the wrapping case on an intermediate buffer when the element boundary doesn't line up with the end of the buffer (because a previous field in the same transfer was a smaller size).

**Comment by streichler** [comment]
I think this should be fixed with 860d294.  @magnatelee can you try any other cases that were showing this problem as well?

**Comment by magnatelee** [comment]
The issue has been fixed in all cases I'm concerned with.
